app/actions/deployment_create.rb

```diff
@@ -83,7 +83,25 @@ def create(app:, user_audit_info:, message:)
 
         deployment
       rescue RevisionResolver::NoUpdateRollback, Sequel::ValidationFailed, AppStart::InvalidApp => e
-        error = DeploymentCreate::Error.new(e.message)
+        raise handle_deployment_create_error(e, app)
```

Comment:
isn't there a `raise` inside the method already? I think it's clearer what's going on if you leave it out, but then maybe remove the `raise` on L106 and rename the method to something like `enhanced_deployment_create_error`

Reply:
It think it's also fine to leave it in the method, but probably not in both places